packages/ui/src/kit/VDateInput/VDateInput.vue

```diff
@@ -0,0 +1,228 @@
+
+export default {
+  name: 'VDateInput',
+
+  props: {
+    value: {
+      type: [String, Date],
+      default: '',
+    },
+  },
+
+  data: () => ({
+    innerMonth: 0,
+    innerYear: 2000,
+    isCalendarVisible: false,
+  }),
+
+  computed: {
+    vDateInputCssClass() {
+      return Object.assign({}, this.themeCssClass, {
+        'v-date-input': true,
+      });
+    },
+
+    date() {
+      return dayjs(this.value);
+    },
+
+    formattedInnerMonth() {
+      return dayjs()
+        .month(this.innerMonth)
+        .format('MMMM');
+    },
+
+    currentCalendar() {
+      return getMonthCalendar(dayjs(new Date(this.innerYear, this.innerMonth)));
+    },
+
+    formattedValue() {
+      return this.date.format('YYYY/MM/DD');
+    },
+  },
+
+  watch: {
+    isCalendarVisible(val) {
+      if (val) {
+        document.body.addEventListener('click', this.onClickAnywhere);
+        window.addEventListener('keypress', this.onWindowEscPress);
+      } else {
+        document.body.removeEventListener('click', this.onClickAnywhere);
+        window.removeEventListener('keypress', this.onWindowEscPress);
+      }
+    },
+
+    value(val, oldVal) {
+      const oldDate = dayjs(oldVal);
+      const newDate = dayjs(val);
+      const newDateMonth = newDate.month();
+      const newDateYear = newDate.year();
+
+      if (newDateMonth !== oldDate.month() || newDateYear !== oldDate.year()) {
+        this.innerMonth = newDateMonth;
+        this.innerYear = newDateYear;
+      }
+    },
+  },
+
+  methods: {
+    isSelectedDay({ origin }) {
+      return (
+        this.date.date() === origin.date() &&
+        this.date.month() === origin.month() &&
+        this.date.year() === origin.year()
+      );
+    },
+
+    onBlurInput(e) {
+      console.log(e);
+      const valueDate = dayjs(e.target.value);
+
+      if (valueDate.isValid()) {
+        this.$emit('change', valueDate.toDate());
+      }
+    },
+
+    onFocusInput() {
+      this.isCalendarVisible = true;
+    },
+
+    onChangeYearInput(e) {
+      this.innerYear = e.target.value;
+    },
+
+    onClickPreviousMonth() {
+      if (this.innerMonth === 0) {
+        this.innerMonth = 11;
+        this.innerYear = this.innerYear - 1;
+      } else {
+        this.innerMonth = this.innerMonth - 1;
+      }
+    },
+
+    onClickNextMonth() {
+      if (this.innerMonth === 11) {
+        this.innerMonth = 0;
+        this.innerYear = this.innerYear + 1;
+      } else {
+        this.innerMonth = this.innerMonth + 1;
+      }
+    },
+
+    onClickDay({ origin }) {
+      const { innerMonth, innerYear } = this;
+      const newDate = origin
+        .month(innerMonth)
+        .year(innerYear)
+        .toDate();
+
+      this.$emit('change', newDate);
+      this.isCalendarVisible = false;
+    },
+
+    onWindowEscPress(e) {
+      if (e.key === 'Escape') {
+        this.isCalendarVisible = false;
+      }
+    },
+
+    onClickAnywhere(e) {
+      let currentNode = e.target
```

Comment:
maybe we need create `outside/inside click` component?
as i see, we need not the same, but almost equal logic in `VNavSubMenu.vue`